File: backend/routes/signin.py
# signin.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from backend.database import SessionLocal

router = APIRouter()

def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

@router.post("/")
def sign_in_worker():
    return {"message": "Successfully signed in. Thank You"}

File: backend/routes/signin.js
import express from "express";
import pool from "../config/db.js";
import { signIn, signOut, getAttendanceRecords } from "../models/attendance.js";


const router = express.Router();

router.post("/", async (req, res) => {
  const { name, company, site, swms, induction } = req.body;

  if (!name || !company || !site) {
    return res.status(400).json({ error: "Missing required fields" });
  }

  try {
    const sql = `INSERT INTO attendance (name, company, site, status, swms, induction) VALUES (?, ?, ?, 'IN', ?, ?)`;
    await pool.query(sql, [name, company, site, swms, induction]);

    res.json({ message: "Successfully signed in" });
  } catch (error) {
    res.status(500).json({ error: "Database error" });
  }
});

export default router;
